### ISSUE TITLE
[HOLD for payment 2025-01-02] Debug Mode - "Create" button flash after creating a violation

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53027]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.66-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Can't check as no `Debug Mode` on production
**Email or phone of affected tester (no customers):** dave0123seife@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Enable debug mode in Troubleshoot
3. Open an expense thread (create one if you don't have any)
4. Enter debug view then check on "View transaction" tab
5. Go to "violations" > Create a violation and save it

## Expected Result:
"Create" button does not flash after creating a violation

## Actual Result:
"Create" button flash after creating a violation

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/1989bf8e-abcd-4ee1-8774-dfea86edd3ce

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @slafortune</details>

**Comment by DylanDylann** [comment]
cc @pac-guerreiro

**Comment by mountiny** [comment]
This is a debug feature only so not a blocker. @DylanDylann @pac-guerreiro please follow up on this one from your PR cc @puneetlath in case you want to take over as a CME too cc @fabioh8010

**Comment by pac-guerreiro** [comment]
Hi! I’m Pedro Guerreiro from Callstack - expert contributor group. I’d like to work on this task!

**Comment by pac-guerreiro** [comment]
@IuliiaHerets I wasn't able to reproduce this behaviour on latest main, can you test it again? 😄 

https://github.com/user-attachments/assets/d837323c-94f7-4e82-8f11-77f7d1f372ce

**Comment by DylanDylann** [comment]
I still can reproduce this bug. @pac-guerreiro Could you please try again with some consecutive create?

https://github.com/user-attachments/assets/57efef6e-9d5b-4dc0-9579-144dda9bb3bc

**Comment by DylanDylann** [comment]
Another bug I see on the debug feature:

https://github.com/Expensify/App/blob/3f4e93ee40462b42f5fe331fdf9db5c9bbb1a421/src/pages/Debug/ReportAction/DebugReportActionCreatePage.tsx#L117

we use the set method but we don't add the previous data. It caused when adding new report action, all previous report actions will be removed

**Comment by IuliiaHerets** [comment]
@pac-guerreiro Tester also can still reproduce the issue in the latest build 

https://github.com/user-attachments/assets/973b9717-5543-42eb-9087-d8e8eab582a1

**Comment by pac-guerreiro** [comment]
@IuliiaHerets @DylanDylann 

I'll try it again, but it's not an easy one 😅 Could you also try creating a report action to confirm if it also happens?

**Comment by pac-guerreiro** [comment]
Today I didn't have the time to test this again. Tomorrow I'll take another look at it!

**Comment by IuliiaHerets** [comment]
@pac-guerreiro Issue is reproducible, build v9.0.67-1

https://github.com/user-attachments/assets/648b5c2c-67b5-4496-ad21-cf6a967b8099

**Comment by JKobrynski** [comment]
Hi, I'm Julian from Callstack and I'm going to work on this issue in Pedro's absence.

**Comment by JKobrynski** [comment]
Today I tested a solution which gets rid of the **ScrollView** that wrapped the **FlatList** with `scrollEnabled={false}`, so there is only one scrollable component, I also wrapped `renderItem` method with **useCallback**, but none of these solutions fixed that flashing effect. 

My plan for the further investigation is to check how this component is rendered in **DebugTransactionPage** and look for the source of this bug there.

**Comment by JKobrynski** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

"Create" button is flashing after creating a new transaction violation in debug mode.

### What is the root cause of that problem?

According to my investigation, this is caused by **FlatList**. Even replacing it with **FlashList** doesn't help. To me, it looked like **FlashList** made it occur less often, but I might be wrong, and it doesn't really matter as the bug still reproduces.

### What changes do you think we should make in order to solve the problem?


As the **<DebugTransactionViolations />** contains a **ScrollView** that wraps a **FlatList** with `scrollEnabled={false}` we can get rid of the **FlatList** component and replace it with a standard `data.map((item) => renderItem(item))` solution. This is not one of the super long and data heavy lists, so removing the non-scrollable **FlatList** should not impact the performance. I tested this solution and I couldn't reproduce the flashing effect anymore. It was also still as smooth as before.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


N/A

### What alternative solutions did you explore? (Optional)

As mentioned above, I tried replacing `FlatList` with `FlashList` but it didn't fix the issue. It seemed to me like the issue happens less often, but it didn't fix it.

**Comment by blimpich** [comment]
Weird bug. I'm good with the above solution if we make sure to add a comment explaining why we made the changes we did. Seems like a bug that wouldn't really be able to be prevented with a unit test and also would be easy to forget about and have a regression come up later.

**Comment by JKobrynski** [comment]
The [PR](https://github.com/Expensify/App/pull/53655) is ready for review!

**Comment by pac-guerreiro** [comment]
@DylanDylann @IuliiaHerets I'm back from my vacation, so I'll take it from here 😄

**Comment by pac-guerreiro** [comment]
I'm able to reproduce this issue on track/submit expense flow as well:

https://github.com/user-attachments/assets/4eb10a22-351f-4ffe-b3bf-93f9815fdaa6

I suspect this issue is caused by `OnyxTabNavigator`. I just asked internally if someone has experienced this before and can help with it 😄

**Comment by pac-guerreiro** [comment]
@puneetlath @IuliiaHerets 

Yesterday I addressed the eslint errors on the PR but it seems that they still persist 😅 

I'll address them in a moment and let you know!

**Comment by pac-guerreiro** [comment]
@puneetlath @IuliiaHerets 

All remaining issues addressed! Now waiting on @DylanDylann to do his magic 🪄

**Comment by pac-guerreiro** [comment]
@puneetlath @IuliiaHerets 

Today I resolved all conflicts in the PR 😄

**Comment by pac-guerreiro** [comment]
@puneetlath @IuliiaHerets 

Today I fixed the app crash reported by @DylanDylann!

I'll be away until January 2nd but someone should take care of my work 😄 Happy holidays! 🎉

**Comment by puneetlath** [comment]
@DylanDylann does this require payment or is it a regression from a previous PR?

**Comment by pac-guerreiro** [comment]
@puneetlath this issue is not a regression 😉

**Comment by DylanDylann** [comment]
> @DylanDylann does this require payment 

I think yes

**Comment by slafortune** [comment]
@puneetlath do you have a milestone tracking that you are paying all these GH's related to debug mode?

**Comment by puneetlath** [comment]
Yes I do. Paid here: https://www.upwork.com/nx/wm/workroom/38901941

@DylanDylann I think we're done with the debug mode stuff and I can close that Upwork job out now. Is that right?

**Comment by DylanDylann** [comment]
All done. Let's close this issue

**Comment by puneetlath** [comment]
@DylanDylann shall I also close the Upwork job out now?

**Comment by DylanDylann** [comment]
Yess

**Comment by puneetlath** [comment]
Ok, all done. Thanks y'all!
